common/types/chain_receipt.go

```diff
@@ -0,0 +1,14 @@
+package types
+
+import "math/big"
+
+type ChainReceipt[TX_HASH Hashable, BLOCK_HASH Hashable] interface {
```

Comment:
Why do we call this a `ChainReceipt` rather than a `TxReceipt` or even just `Receipt`?